runtime/src/bank.rs

```diff
@@ -3859,37 +3867,52 @@ impl Bank {
-                let execution_details = processed_tx.execution_details;
-                let LoadedTransaction {
-                    rent_debits,
-                    accounts: loaded_accounts,
-                    loaded_accounts_data_size,
-                    fee_details,
-                    ..
-                } = processed_tx.loaded_transaction;
-
-                // Rent is only collected for successfully executed transactions
-                let rent_debits = if execution_details.was_successful() {
-                    rent_debits
-                } else {
-                    RentDebits::default()
-                };
+            .map(|processing_result| match processing_result? {
+                ProcessedTransaction::Executed(executed_tx) => {
+                    let execution_details = executed_tx.execution_details;
+                    let LoadedTransaction {
+                        rent_debits,
+                        accounts: loaded_accounts,
+                        loaded_accounts_data_size,
+                        fee_details,
+                        ..
+                    } = executed_tx.loaded_transaction;
+
+                    // Rent is only collected for successfully executed transactions
+                    let rent_debits = if execution_details.was_successful() {
+                        rent_debits
+                    } else {
+                        RentDebits::default()
+                    };
 
-                Ok(CommittedTransaction {
-                    status: execution_details.status,
-                    log_messages: execution_details.log_messages,
-                    inner_instructions: execution_details.inner_instructions,
-                    return_data: execution_details.return_data,
-                    executed_units: execution_details.executed_units,
-                    fee_details,
-                    rent_debits,
+                    Ok(CommittedTransaction {
+                        status: execution_details.status,
+                        log_messages: execution_details.log_messages,
+                        inner_instructions: execution_details.inner_instructions,
+                        return_data: execution_details.return_data,
+                        executed_units: execution_details.executed_units,
+                        fee_details,
+                        rent_debits,
+                        loaded_account_stats: TransactionLoadedAccountsStats {
+                            loaded_accounts_count: loaded_accounts.len(),
+                            loaded_accounts_data_size,
+                        },
+                    })
+                }
+                ProcessedTransaction::FeesOnly(fees_only_tx) => Ok(CommittedTransaction {
+                    status: Err(fees_only_tx.load_error),
+                    log_messages: None,
+                    inner_instructions: None,
+                    return_data: None,
+                    executed_units: 0,
```

Comment:
@tao-stones can you confirm this understanding?

These `executed_units` in `CommittedTransaction` are supposed to be the builtin + user-space costs.
These do **not** include signature costs, write-lock costs, etc.

Basically I want to make sure that we cannot include an **infinite** number of fee-only transactions. There should be some cost towards block-limits even if it is smaller than if it were actually executed.

Reply:
> These executed_units in CommittedTransaction are supposed to be the builtin + user-space costs.
These do not include signature costs, write-lock costs, etc.

Correct, it's the actual units consumed during execution. `0` is proper for not executed. In this case, cost tracker will adjust down `programs_execution_cost`, but other transaction costs (signature, write-locks etc) still count towards limits.